libraries/fc/include/fc/io/raw.hpp

```diff
@@ -142,9 +142,26 @@ namespace fc {
 
     template<typename Stream, typename T, size_t N>
     inline void pack( Stream& s, const fc::array<T,N>& v) {
-      s.write((const char*)&v.data[0],N*sizeof(T));
+       for (uint64_t i = 0; i < N; ++i)
```

Comment:
We should probably not remove this optimization for lots of fundamental types.

Can we guard this something like `std::is_scalar && ! std::is_pointer` so that the slow path is only used when it will probably differ from the fast path?

Reply:
we use is_fundemental, isn't that the one I should use, since we don't support packing a pointer (at least not directly)

Reply:
`is_fundamental` won't catch pointers, enumerations pointers-to-members etc.  we have a change incoming to other uses of `is_fundamental` to change them to `is_scalar` and either `static_assert` if pointers are invalid.   It is more relevant when we are using "magic_get" because using the `for_each` on something like a pointer tends to cause an infinite recursion (if the call path into and out of for_each is recursive).

Reply:
done